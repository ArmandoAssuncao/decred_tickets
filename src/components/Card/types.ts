import { ITicket } from 'store/ticket/types'

export interface IProps {
    onPress?: () => void
    type: ITicket['status']
    value: number
}

export interface IState {
}
