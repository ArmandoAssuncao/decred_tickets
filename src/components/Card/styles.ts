import { ViewStyle } from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'

export default EStyleSheet.create({
    container: {
        backgroundColor: '$secondColor',
        padding: '0.5rem',
        margin: '0.2rem',
        flexDirection: 'row',
        borderRadius: 5,
        flex: 1,
    } as ViewStyle,
    image: {
        height: 50,
        width: 50,
    } as ViewStyle,
    imageContainer: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
    } as ViewStyle,
    textContainer: {
        flex: 2,
        alignItems: 'center',
        justifyContent: 'center',
    } as ViewStyle,
})
