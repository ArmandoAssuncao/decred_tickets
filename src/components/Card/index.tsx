import React from 'react'
import { Image, View, TouchableOpacity } from 'react-native'

import { IProps, IState } from './types'
import styles from './styles'

import i18n from 'config/i18n'
import { Text } from 'components'

export default class Card extends React.Component<IProps, IState> {
    public static defaultProps: Partial<IProps> = {
        value: 0,
    }

    render(): JSX.Element {
        const { onPress, type, value } = this.props
        let title = null
        let img = null
        switch (type) {
            case 'revoked':
                img = require('@assets/imgs/ticket-revoked.png')
                title = i18n.t('decred.ticket.revoked', { defaultValue: 'Revoked' })
                break
            case 'voted':
                img = require('@assets/imgs/ticket-voted.png')
                title = i18n.t('decred.ticket.voted', { defaultValue: 'Voted' })
                break
            case 'expired':
                img = require('@assets/imgs/ticket-expired.png')
                title = i18n.t('decred.ticket.expired', { defaultValue: 'Expired' })
                break
            case 'missed':
                img = require('@assets/imgs/ticket-missed.png')
                title = i18n.t('decred.ticket.missed', { defaultValue: 'Missed' })
                break
            case 'unmined':
                img = require('@assets/imgs/ticket-unmined.png')
                title = i18n.t('decred.ticket.unmined', { defaultValue: 'Unmined' })
                break
            case 'immature':
                img = require('@assets/imgs/ticket-immature.png')
                title = i18n.t('decred.ticket.immature', { defaultValue: 'Immature' })
                break
            case 'live':
                img = require('@assets/imgs/ticket-live.png')
                title = i18n.t('decred.ticket.live', { defaultValue: 'Live' })
                break
            default:
                break
        }
        return (
            <TouchableOpacity onPress={onPress} style={styles.container}>
                <View style={styles.imageContainer}>
                    <Image
                        style={styles.image}
                        source={img}
                    />
                </View>
                <View style={styles.textContainer}>
                    <Text m={'Live'} size="md" type="bold">{title}</Text>
                    <Text m={value.toString()} size="xxl" type="bold">{value.toString()}</Text>
                </View>
            </TouchableOpacity>
        )
    }
}
