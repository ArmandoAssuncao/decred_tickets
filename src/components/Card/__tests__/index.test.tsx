import React from 'react'
import 'react-native'
import { shallow } from 'enzyme'

import Card from '../index'

describe('Card', () => {
    const wrapper = shallow(<Card type="live" value={10} onPress={jest.fn()} />)

    it('match snapshot when prop type is "live"', () => {
        wrapper.setProps({type: 'live'})
        expect(wrapper).toMatchSnapshot()
    })

    it('match snapshot when prop type is "revoked"', () => {
        wrapper.setProps({type: 'revoked'})
        expect(wrapper).toMatchSnapshot()
    })

    it('match snapshot when prop type is "voted"', () => {
        wrapper.setProps({type: 'voted'})
        expect(wrapper).toMatchSnapshot()
    })

    it('match snapshot when prop type is "expired"', () => {
        wrapper.setProps({type: 'expired'})
        expect(wrapper).toMatchSnapshot()
    })

    it('match snapshot when prop type is "missed"', () => {
        wrapper.setProps({type: 'missed'})
        expect(wrapper).toMatchSnapshot()
    })

    it('match snapshot when prop type is "unmined"', () => {
        wrapper.setProps({type: 'unmined'})
        expect(wrapper).toMatchSnapshot()
    })

    it('match snapshot when prop type is "immature"', () => {
        wrapper.setProps({type: 'immature'})
        expect(wrapper).toMatchSnapshot()
    })

    it('has 3 Component component', () => {
        expect(wrapper.find('Component')).toHaveLength(3)
    })

    it('has 2 Text component', () => {
        expect(wrapper.find('Text')).toHaveLength(2)
    })

    it('has 1 Image component', () => {
        expect(wrapper.find('Image')).toHaveLength(1)
    })

    describe('actions', () => {
        it('call onPress when pressed', () => {
            wrapper.props().onPress()
            expect(wrapper.prop('onPress')).toBeCalled()
        })
    })
})
