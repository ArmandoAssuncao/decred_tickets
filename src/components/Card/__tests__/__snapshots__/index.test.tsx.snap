// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card match snapshot when prop type is "expired" 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#F3F6F6",
      "borderRadius": 5,
      "flex": 1,
      "flexDirection": "row",
      "margin": 3.2,
      "padding": 3.2,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      m="Live"
      size="md"
      type="bold"
    >
      Expired
    </Text>
    <Text
      m="10"
      size="xxl"
      type="bold"
    >
      10
    </Text>
  </Component>
</TouchableOpacity>
`;

exports[`Card match snapshot when prop type is "immature" 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#F3F6F6",
      "borderRadius": 5,
      "flex": 1,
      "flexDirection": "row",
      "margin": 3.2,
      "padding": 3.2,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      m="Live"
      size="md"
      type="bold"
    >
      Immature
    </Text>
    <Text
      m="10"
      size="xxl"
      type="bold"
    >
      10
    </Text>
  </Component>
</TouchableOpacity>
`;

exports[`Card match snapshot when prop type is "live" 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#F3F6F6",
      "borderRadius": 5,
      "flex": 1,
      "flexDirection": "row",
      "margin": 3.2,
      "padding": 3.2,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      m="Live"
      size="md"
      type="bold"
    >
      Live
    </Text>
    <Text
      m="10"
      size="xxl"
      type="bold"
    >
      10
    </Text>
  </Component>
</TouchableOpacity>
`;

exports[`Card match snapshot when prop type is "missed" 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#F3F6F6",
      "borderRadius": 5,
      "flex": 1,
      "flexDirection": "row",
      "margin": 3.2,
      "padding": 3.2,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      m="Live"
      size="md"
      type="bold"
    >
      Missed
    </Text>
    <Text
      m="10"
      size="xxl"
      type="bold"
    >
      10
    </Text>
  </Component>
</TouchableOpacity>
`;

exports[`Card match snapshot when prop type is "revoked" 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#F3F6F6",
      "borderRadius": 5,
      "flex": 1,
      "flexDirection": "row",
      "margin": 3.2,
      "padding": 3.2,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      m="Live"
      size="md"
      type="bold"
    >
      Revoked
    </Text>
    <Text
      m="10"
      size="xxl"
      type="bold"
    >
      10
    </Text>
  </Component>
</TouchableOpacity>
`;

exports[`Card match snapshot when prop type is "unmined" 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#F3F6F6",
      "borderRadius": 5,
      "flex": 1,
      "flexDirection": "row",
      "margin": 3.2,
      "padding": 3.2,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      m="Live"
      size="md"
      type="bold"
    >
      Unmined
    </Text>
    <Text
      m="10"
      size="xxl"
      type="bold"
    >
      10
    </Text>
  </Component>
</TouchableOpacity>
`;

exports[`Card match snapshot when prop type is "voted" 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#F3F6F6",
      "borderRadius": 5,
      "flex": 1,
      "flexDirection": "row",
      "margin": 3.2,
      "padding": 3.2,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      m="Live"
      size="md"
      type="bold"
    >
      Voted
    </Text>
    <Text
      m="10"
      size="xxl"
      type="bold"
    >
      10
    </Text>
  </Component>
</TouchableOpacity>
`;
